libevm/VM.cpp

```diff
@@ -151,764 +136,745 @@ bytesConstRef VM::execImpl(u256& io_gas, ExtVMFace& _ext, OnOpFunc const& _onOp)
-			m_newMemSize = memNeed(*m_SP, *(m_SP - 2));
+		CASE_BEGIN(CALLDATACOPY)
+			m_copyMemSize = toUint64(*(SP - 2));
+			m_newMemSize = memNeed(*SP, *(SP - 2));
 			updateMem();
 			onOperation();
 			updateIOGas();
 
-			copyDataToMemory(m_ext->data, m_SP);
-			break;
+			copyDataToMemory(m_ext->data, SP);
+			++PC;
+		CASE_END
 
-		case Instruction::CODECOPY:
-			m_copyMemSize = toUint64(*(m_SP - 2));
-			m_newMemSize = memNeed(*m_SP, *(m_SP - 2));
+		CASE_BEGIN(CODECOPY)
+			m_copyMemSize = toUint64(*(SP - 2));
+			m_newMemSize = memNeed(*SP, *(SP - 2));
 			updateMem();
 			onOperation();
 			updateIOGas();
 
-			copyDataToMemory(&m_ext->code, m_SP);
-			break;
+			copyDataToMemory(&m_ext->code, SP);
+			++PC;
+		CASE_END
 
-		case Instruction::EXTCODECOPY:
+		CASE_BEGIN(EXTCODECOPY)
 		{
-			m_copyMemSize = toUint64(*(m_SP - 3));
-			m_newMemSize = memNeed(*(m_SP - 1), *(m_SP - 3));
+			m_copyMemSize = toUint64(*(SP - 3));
+			m_newMemSize = memNeed(*(SP - 1), *(SP - 3));
 			updateMem();
 			onOperation();
 			updateIOGas();
 
-			auto a = asAddress(*m_SP);
-			--m_SP;
-			copyDataToMemory(&m_ext->codeAt(a), m_SP);
-			break;
+			auto a = asAddress(*SP);
+			--SP;
+			copyDataToMemory(&m_ext->codeAt(a), SP);
+			++PC;
 		}
+		CASE_END
+
+
+		CASE_BEGIN(GASPRICE)
+			onOperation();
+			updateIOGas();
 
-		case Instruction::GASPRICE:
+			*++SP = m_ext->gasPrice;
+			++PC;
+		CASE_END
+
+		CASE_BEGIN(BLOCKHASH)
 			onOperation();
 			updateIOGas();
 
-			*++m_SP = m_ext->gasPrice;
-			break;
+			*SP = (u256)m_ext->blockHash(*SP);
+			++PC;
+		CASE_END
 
-		case Instruction::BLOCKHASH:
+		CASE_BEGIN(COINBASE)
 			onOperation();
 			updateIOGas();
 
-			*m_SP = (u256)m_ext->blockHash(*m_SP);
-			break;
+			*++SP = (u160)m_ext->envInfo().author();
+			++PC;
+		CASE_END
 
-		case Instruction::COINBASE:
+		CASE_BEGIN(TIMESTAMP)
 			onOperation();
 			updateIOGas();
 
-			*++m_SP = (u160)m_ext->envInfo().author();
-			break;
+			*++SP = m_ext->envInfo().timestamp();
+			++PC;
+		CASE_END
 
-		case Instruction::TIMESTAMP:
+		CASE_BEGIN(NUMBER)
 			onOperation();
 			updateIOGas();
 
-			*++m_SP = m_ext->envInfo().timestamp();
-			break;
+			*++SP = m_ext->envInfo().number();
+			++PC;
+		CASE_END
 
-		case Instruction::NUMBER:
+		CASE_BEGIN(DIFFICULTY)
 			onOperation();
 			updateIOGas();
 
-			*++m_SP = m_ext->envInfo().number();
-			break;
+			*++SP = m_ext->envInfo().difficulty();
+			++PC;
+		CASE_END
 
-		case Instruction::DIFFICULTY:
+		CASE_BEGIN(GASLIMIT)
 			onOperation();
 			updateIOGas();
 
-			*++m_SP = m_ext->envInfo().difficulty();
-			break;
+			*++SP = m_ext->envInfo().gasLimit();
+			++PC;
+		CASE_END
 
-		case Instruction::GASLIMIT:
+		CASE_BEGIN(POP)
 			onOperation();
 			updateIOGas();
 
-			*++m_SP = m_ext->envInfo().gasLimit();
-			break;
+			--SP;
+			++PC;
+		CASE_END
 
-		case Instruction::POP:
+		CASE_BEGIN(PUSHC)
 			onOperation();
 			updateIOGas();
 
-			--m_SP;
-			break;
+			*++SP = m_pool[m_code[++PC]];
+			++PC;
```

Comment:
I think it's wrong, thank s.  Still debugging this code, and it should be inside macros.